src/test/java/spoon/test/prettyprinter/DefaultPrettyPrinterTest.java

```diff
@@ -234,4 +237,22 @@ public void useFullyQualifiedNamesInCtElementImpl_toString() throws Exception {
 		;
 		assertEquals( "the toString method of CtElementImpl should not shorten type names as it has no context or import statements", expected, computed );
 	}
+
+	@Test
+	public void printClassCreatedWithSpoon() throws Exception {
+
+		/* test that spoon is able to print a class that he created without setting manually the output (default configuration) */
+
+		Launcher l = new Launcher();
+		l.getEnvironment().setNoClasspath(true);
+		l.buildModel();
+		Factory factory = l.getFactory();
+		CtClass<Object> ctClass = factory.Class().create("foo.Bar");
+		PrettyPrinter pp = new DefaultJavaPrettyPrinter(factory.getEnvironment());
+		JavaOutputProcessor jop =
+				new JavaOutputProcessor(File.createTempFile("foo","").getParentFile(),pp);
+		jop.setFactory(factory);
+		jop.createJavaFile(ctClass);
+		assertTrue(new File("/tmp/foo/Bar.java").exists());
```

Comment:
`File#createTempFile` will create the `Bar` class in the `/tmp` directory in all systems (Linux, MacOS and Windows)? I have doubts. (: